docs/architecture/adr-003-subaccounts.md

```diff
@@ -0,0 +1,102 @@
+# ADR 3: Module SubAccounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon Supply Keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+We want to support the ability to define and manage sub-module-accounts.
+
+## Decision
+
+We will use the type `ModuleMultiAccount` to manage sub-accounts of type `ModuleAccount`.
+A `ModuleMultiAccount` may have zero or more `ModuleAccount`s.
+Each sub-account will utilize the existing permissioned properties of `ModuleAccount`.
+`ModuleMultiAccount` and `ModuleAccount` have no pubkeys.
+There is no limit on the number of `ModuleAccount`s that a `ModuleMultiAccount` can have.
+A `ModuleMultiAccount` has no permissions since it cannot hold any coins.
+Its constructor returns a `ModuleMultiAccount` with no `ModuleAccount`s.
+A sub-account cannot be removed from a `MultiModuleAccount`.
+The account number assigned to sub-accounts will begin at 0 and be monotonically auto incrementing.
+
+### Implementation Changes
+
+Introduce a new type into `x/supply`:
+
+* `ModuleMultiAccount`
+
+```go
+// Implements the Account interface.
+// SetCoins will return an error to prevent ModuleMultiAccount address from having a balance.
+// ModuleAccounts are appended to the SubAccounts array.
+// Passively tracks the sum of all ModuleAccount balances.
+type ModuleMultiAccount struct {
+    SubAccounts []ModuleAccount
+    Coins sdk.Coins // passively track all sub account balances
+
+    CreateSubAccount(name string, permissions ...string) int // returns account number of sub-account
```

Comment:
This should probably return an error under the scenario that the subaccount is attempted to be created with non-inherited permissions of the `ModuleMultiAccount`